internal/gcsutil/util.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package gcsutil contains utility functions for Google Cloud Storage data access.
+package gcsutil
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+)
+
+// GCSClient contains a Google Cloud Storage client.
+type GCSClient struct {
+	Client *storage.Client
+}
+
+// NewClient creates and returns a new GCSClient.
+func NewClient(ctx context.Context) (*GCSClient, error) {
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("could not create a new Google Cloud Storage client: %v", err)
+	}
+	defer client.Close()
```

Comment:
Would we do that here? Would we not ask the owner of the client to close it? I am not sure. What do you think?

Reply:
You mean close it in main? I think "It is recommended to close immediately after checking the error" that's why I kept it here.

Reply:
Sounds good!